src/main/java/com/getindata/connectors/http/sink/httpclient/JavaNetSinkHttpClient.java

```diff
@@ -0,0 +1,111 @@
+package com.getindata.connectors.http.sink.httpclient;
+
+import com.getindata.connectors.http.SinkHttpClient;
+import com.getindata.connectors.http.SinkHttpClientResponse;
+import com.getindata.connectors.http.sink.HttpSink;
+import com.getindata.connectors.http.sink.HttpSinkRequestEntry;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * An implementation of {@link SinkHttpClient} that uses Java 11's {@link HttpClient}.
+ */
+public class JavaNetSinkHttpClient implements SinkHttpClient {
+  private final HttpClient httpClient;
+
+  public JavaNetSinkHttpClient() {
+    this.httpClient = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
+  }
+
+  /**
+   * A wrapper structure around an arbitrary element, keeping a reference to a particular
+   * {@link HttpSinkRequestEntry}. Used internally by the {@code HttpSinkWriter} to pass
+   * {@code HttpSinkRequestEntry} along some other element that is logically connected with it
+   * (e.g., full HTTP request built from the {@code HttpSinkRequestEntry}).
+   *
+   * @param <T>
+   */
+  @RequiredArgsConstructor
+  @EqualsAndHashCode
+  private static class HttpSinkRequestEntryWrapper<T> {
```

Comment:
will other implementations of `SinkHttpClient`  would have to use this wrapper?
If yes, then it should be separated class.

Reply:
Sorry but this class is used to wrap a request or response?

Reply:
It's rather used internally by `JavaNet` implementation. It wraps either of them. It's again kind of a namespace clash — we've got `HttpSinkRequest` which gets created in `elementConverter` and is internally passed through the Sink, and `HttpRequest`/`HttpResponse` classes coming from `java.net.http` package which are used by Java Net's `HttpClient`. And this particular class wraps `HttpSinkRequest` along with either of `java.net.http` classes.